.storybook/blocks/ComponentDetails.jsx

```diff
@@ -264,13 +306,101 @@ function fetchNpmData(packageName, setnpmData, setIsLoading) {
 	}, [cache, setCache, packageName, setnpmData, setIsLoading]);
 }
 
+const fetchLogo = (brand) => {
+	switch(brand) {
+		case "npm":
+			return NpmSVG;
+		case "GitHub":
+			return GitHubSVG;
+		case "Adobe":
+			return AdobeSVG;
+	}
+
+	return;
+}
+
+/**
+ * Displays a resource card containing text and an image that links to a particular resource.
+ * 
+ * @param {string} heading - heading of the resource card
+ * @param {string} alt - additional description of the resource card
+ * @param {string} image - the SVG image
+ * @param {string} href - optional link to the resource, found in packageJson?.spectrum?.guidelines
+ * @returns {string}
+ */
+export const ResourceLinkContent = ({ heading, alt, logo, href }) => {
+	if (!href) return;
+
+	return (
+		<ResourceLink href={href} className="sb-unstyled" title={alt}>
+			<ResourceIconWrapper dangerouslySetInnerHTML={{ __html: fetchLogo(logo) }} />
+			<ResourceTextWrapper>
+				{heading ? <Heading size="xs">{heading}</Heading> : ""}
+				{alt ? <Body size="s">{alt}</Body> : ""}
+			</ResourceTextWrapper>
+		</ResourceLink>
+	);
+};
+
+/**
+ * Displays the list of relevant component links (to NPM, repo, guidelines, etc).
+ * 
+ * The rootClassName is read from the story's default args, found in the story's metadata.
+ * 
+ * The for loop is particularly helpful to match guidelines links for any nested components
+ * (i.e. meter, form). We need to check that the rootClassName matches the rootClass found
+ * in the packageJson.spectrum, to link to the correct guidelines page.
+ * 
+ * @param {string} packageName - packageName sourced from packageJson?.name
+ * @param {string[]} spectrumData - an array of objects sourced from packageJson?.spectrum
+ * @param {string} rootClassName - a component's default rootClass arg 
+ * @returns {string}
+ */
+export const ResourceListDetails = ({ packageName, spectrumData = [], rootClassName }) => {
+	if (!packageName) return;
```

Comment:
Out of curiosity, are there any components (or other cases) where there's no `packageName`?

Reply:
I don't think there are, at least purposely. But I do think that actually helped us find a bug earlier! 

`packageName` is getting set in the parameters in each of the story files (in tandem with `import packageJson from "../package.json";`). I noticed that progress bar and switch were not returning the `ComponentDetails` or the `TaggedReleases` components. About a week or two ago, Patrick figured out that the import and the parameters weren't in those components' files (so `packageName` didn't exist). 

As soon as he fixed that, the switch/progressbar `packageName` and all the data came in correctly, and `ComponentDetails` and `TaggedReleases` finally rendered 🥳